code/modules/hippie/items/weapons/hockey.dm

```diff
@@ -0,0 +1,298 @@
+	throwforce_mult = 1.4 //More powerful than a baseball bat because it costs 20 fucking tc and more or less denies you any ranged option.
+	specthrow_maxwclass = 4 //If you have the time to build a spear on the fly with no backpack to keep the bits in you deserve to be able to chuck it around freely.
+	specthrowmsg = "chipped"
+	sharpness = IS_SHARP_ACCURATE //Very sharp to make up for the comparativly low damage.
+	var/click_delay = 1.3
+	var/deflect_chance = 50
+
+	var/obj/item/weapon/hockeypack/pack
+
+/obj/item/weapon/twohanded/hockeystick/update_icon()
+	icon_state = "hockeystick[wielded]"
+	return
+
+/obj/item/weapon/twohanded/hockeystick/New(parent_pack)
+	..()
+	if(check_pack_exists(parent_pack, src))
+		pack = parent_pack
+		loc = pack
+
+	return
+
+
+/obj/item/weapon/twohanded/hockeystick/attack(mob/living/target, mob/living/user) //Sure it's the powerfist code, right down to the sound effect. Gonna be fun though.
+
+	if(!wielded)
+		return ..()
+
+	target.apply_damage(force, BRUTE)	//If it's a mob but not a humanoid, just give it plain brute damage.
+
+	target.visible_message("<span class='danger'>[target.name] was pucked by [user] 'eh!</span>", \
+		"<span class='userdanger'>You hear a loud crack 'eh!</span>", \
+		"<span class='italics'>You hear the sound of bones crunching 'eh!</span>")
+
+	var/atom/throw_target = get_edge_target_turf(target, get_dir(src, get_step_away(target, src)))
+	target.throw_at(throw_target, 10, 1)	//Throws the target 10 tiles
+
+	playsound(loc, 'sound/weapons/resonator_blast.ogg', 50, 1)
+
+	add_logs(user, target, "used a hockey stick on", src) //Very unlikeley non-antags are going to get their hands on this but just in case...
+
+	user.changeNext_move(CLICK_CD_MELEE * click_delay)
+
+
+	return
+
+/obj/item/weapon/twohanded/hockeystick/dropped(mob/user) //The Stick is undroppable but just in case they lose an arm better put this here.
+		..()
+		user << "<span class='notice'>The stick is drawn back to the backpack 'eh!</span>"
+		pack.on = 0
+		loc = pack
+
+
+/proc/check_pack_exists(parent_pack, mob/living/carbon/human/M, obj/O)
+	if(!parent_pack || !istype(parent_pack, /obj/item/weapon/hockeypack))
+		qdel(O)
+		return 0
+	else
+		return 1
+
+/obj/item/weapon/twohanded/hockeystick/Move()
+	..()
+	if(loc != pack.loc)
+		loc = pack.loc
+
+/obj/item/weapon/twohanded/hockeystick/IsReflect()
+	if(wielded)
+		if(prob(deflect_chance))
+			return 1
+
+/obj/item/weapon/storage/belt/hockey
+	name = "Holopuck Generator"
+	desc = "A Belt mounted device that quickly fabricates hard-light holopucks that when thrown will stall and slow down foes dealing minor damage. Has a pouch to store a pair of spare pucks"
+	icon_state = "hockey_belt"
+	item_state = "hockey_belt"
+	actions_types = list(/datum/action/item_action/make_puck)
+	storage_slots = 2
+	flags = NODROP
+	can_hold = list(/obj/item/holopuck)
+	var/recharge_time = 100
+	var/charged = TRUE
+	var/timerid
```

Comment:
this is never checked on Destroy, this is also no longer needed since `qdel` now checks for timers